tz/digest.go

```diff
@@ -122,3 +126,50 @@ func mulBitRightGeneric(c00, c10, c01, c11 *GF127, bit bool, tmp *GF127) {
 		*c11 = *tmp
 	}
 }
+
+// MarshalBinary implements [encoding.BinaryMarshaler].
+func (d *digest) MarshalBinary() ([]byte, error) {
+	var (
+		b          = make([]byte, Size)
+		start, end int
+	)
+
+	for i, a := range d.x {
+		state, err := a.MarshalBinary()
+		if err != nil {
+			return nil, err
+		}
+
+		start = gf127.ByteSize * i
+		end = start + gf127.ByteSize
+
+		copy(b[start:end], state)
```

Comment:
Have you benchmarked this against `make([]byte, 0, Size) + append(b, state)`? The latter would be much simpler code.

Reply:
Hm, append works better. Updated

```
BenchmarkMarshalCopy-16         13317152                85.80 ns/op           64 B/op          1 allocs/op
BenchmarkMarshalAppend-16       19068822                63.11 ns/op            0 B/op          0 allocs/op
```